File: 0658-find-k-closest-elements/0658-find-k-closest-elements.java
class Solution {
    class Pair{
        int first;
        int second;

        Pair(int first,int second){
            this.first=first;
            this.second=second;
        }
    }

    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        int n=arr.length;

        // Priority Queue Declaration
        PriorityQueue<Pair> pq=new PriorityQueue<>((a,b)->{
            if(a.first!=b.first){
                // Max wla first lena hai
                return b.first-a.first;
            }
            // Max wala hi second lena hogaaa
            return b.second-a.second;
        });

        for(int i=0;i<n;i++){
            int element=arr[i];

            int dist=Math.abs(element-x);

            Pair curr=new Pair(dist,element);          // Iss trh se raka jaega

            if(pq.size()<k){
                // Still in the Process ; toh randomly push k elements and move fwd
                pq.add(curr);
                continue;
            }

            if(curr.first>pq.peek().first){
                // skip and move fwd
                continue;
            }

            if(curr.first==pq.peek().first && curr.second>=pq.peek().second){
                // Same Distance par bada element hai toh skip
                continue;
            }

            pq.poll();
            pq.add(curr);
        }

        List<Integer> res=new ArrayList<>();           // Taking an Array of the Resultant

        while(!pq.isEmpty()){
            res.add(pq.peek().second);                 // Res Array mein second wley ko push kardo
            pq.poll();                                 // ab pop kardo
        }

        Collections.sort(res);                         // Final Ans Sorted Order mein chahiye

        return res;                                    // Resultant Array return kardo
    }
}

File: 0658-find-k-closest-elements/0658-find-k-closest-elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        n=len(arr)

        # Priority Queue Declaration
        pq=[]

        for i in range(n):
            element=arr[i]

            dist=abs(element-x)

            curr=(-dist,-element)          # Iss trh se raka jaega

            if(len(pq)<k):
                # Still in the Process ; toh randomly push k elements and move fwd
                heapq.heappush(pq,curr)
                continue

            if(dist>-pq[0][0]):
                # skip and move fwd
                continue

            if(dist==-pq[0][0] and element>=-pq[0][1]):
                # Same Distance par bada element hai toh skip
                continue

            heapq.heappop(pq)
            heapq.heappush(pq,curr)

        res=[]                             # Taking an Array of the Resultant

        while(pq):
            res.append(-pq[0][1])          # Res Array mein second wley ko push kardo
            heapq.heappop(pq)              # ab pop kardo

        res.sort()                          # Final Ans Sorted Order mein chahiye

        return res                         # Resultant Array return kardo